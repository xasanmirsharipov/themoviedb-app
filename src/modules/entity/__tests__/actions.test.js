import React from 'react';
import Actions from '../actions';

describe('Entity Actions', () => {

	const entity = 'movie';
	const entityType = 'movieEntityType';

	describe('LoadAll', () => {

		const LoadAll = Actions.LoadAll;

		it('Request', () => {

			const payload = {
				entity,
				name: entityType,
				url: '/movies/popular',
				options: {},
				meta: {
					limit: 10,
					page: 1
				}
			};

			const action = LoadAll.request(payload);

			expect(action).toEqual({
				type: "LOAD_ALL_ENTITY/REQUEST",
				payload
			});

			expect(action).toMatchSnapshot();
		});

		it('Success', () => {

			const payload = {
				entity,
				name: entityType,
				ids: [1,2,3],
				meta: {
					totalCount: 0,
					pageCount: 1,
					currentPage: 1,
					perPage: 20
				}
			};

			const action = LoadAll.success(payload);

			expect(action).toEqual({
				type: "LOAD_ALL_ENTITY/SUCCESS",
				payload
			});

			expect(action).toMatchSnapshot();
		});

		it('Failure', () => {

			const payload = {
				entity,
				name: entityType,
				error: 'error text'
			};

			const action = LoadAll.failure(payload);

			expect(action).toEqual({
				type: "LOAD_ALL_ENTITY/FAILURE",
				payload
			});

			expect(action).toMatchSnapshot();
		});

	});

	describe('LoadOne', () => {

		const LoadOne = Actions.LoadOne;

		it('Request', () => {

			const payload = {
				entity,
				id: 1,
				url: '/movies/popular/1'
			};

			const action = LoadOne.request(payload);

			expect(action).toEqual({
				type: "LOAD_ONE_ENTITY/REQUEST",
				payload
			});

			expect(action).toMatchSnapshot();
		});

		it('Success', () => {

			const payload = {
				entity,
				id: 1
			};

			const action = LoadOne.success(payload);

			expect(action).toEqual({
				type: "LOAD_ONE_ENTITY/SUCCESS",
				payload
			});

			expect(action).toMatchSnapshot();
		});

		it('Failure', () => {

			const payload = {
				entity,
				id: 1,
				error: 'error text'
			};

			const action = LoadOne.failure(payload);

			expect(action).toEqual({
				type: "LOAD_ONE_ENTITY/FAILURE",
				payload
			});

			expect(action).toMatchSnapshot();
		});

	});

});