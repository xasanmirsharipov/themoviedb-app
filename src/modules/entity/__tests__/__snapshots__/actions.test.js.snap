// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entity Actions LoadAll Failure 1`] = `
Object {
  "payload": Object {
    "entity": "movie",
    "error": "error text",
    "name": "movieEntityType",
  },
  "type": "LOAD_ALL_ENTITY/FAILURE",
}
`;

exports[`Entity Actions LoadAll Request 1`] = `
Object {
  "payload": Object {
    "entity": "movie",
    "meta": Object {
      "limit": 10,
      "page": 1,
    },
    "name": "movieEntityType",
    "options": Object {},
    "url": "/movies/popular",
  },
  "type": "LOAD_ALL_ENTITY/REQUEST",
}
`;

exports[`Entity Actions LoadAll Success 1`] = `
Object {
  "payload": Object {
    "entity": "movie",
    "ids": Array [
      1,
      2,
      3,
    ],
    "meta": Object {
      "currentPage": 1,
      "pageCount": 1,
      "perPage": 20,
      "totalCount": 0,
    },
    "name": "movieEntityType",
  },
  "type": "LOAD_ALL_ENTITY/SUCCESS",
}
`;

exports[`Entity Actions LoadOne Failure 1`] = `
Object {
  "payload": Object {
    "entity": "movie",
    "error": "error text",
    "id": 1,
  },
  "type": "LOAD_ONE_ENTITY/FAILURE",
}
`;

exports[`Entity Actions LoadOne Request 1`] = `
Object {
  "payload": Object {
    "entity": "movie",
    "id": 1,
    "url": "/movies/popular/1",
  },
  "type": "LOAD_ONE_ENTITY/REQUEST",
}
`;

exports[`Entity Actions LoadOne Success 1`] = `
Object {
  "payload": Object {
    "entity": "movie",
    "id": 1,
  },
  "type": "LOAD_ONE_ENTITY/SUCCESS",
}
`;
